doc/examples/animation_widget/lib/main.dart

```diff
@@ -14,7 +14,8 @@ SpriteAnimation _animation;
 
 void main() async {
   WidgetsFlutterBinding.ensureInitialized();
-  _sprite = await Sprite.loadSprite('minotaur.png', width: 96, height: 96);
+  final image = await Flame.images.load('minotaur.png');
+  _sprite = Sprite(image, width: 96, height: 96);
```

Comment:
we can have a `await Sprite.load` method that does this as well

Reply:
a static method I mean. not a named constructor. because constructors cant be async

Reply:
I am not sure about that, those helpers will need to duplicate the parameter structure from the constructor and we will need to repeat a bunch of code, on Sprite class that is not that bad, but on Animation for example, where we have different contractors that can lead to a bunch of repeated code.

Actually a have a nice idea to make up for not having those constructors, I will let you know when I have implemented then.

Reply:
I don't think replicating the parameters is a big deal... Animation class has like 100 constructors that are extremely similar. I don't even know what variableSequenced is for example haha. and it already is a copy of the Animation constructors. so one more is not a problem if it's going to simplify use case for some people

Reply:
No sure I agree with that, on Animation class we have the variableSenquence and sequence, and that constructor was almost identical, and that lead us to so problems. Anyway, now with the asset loading on the game, the static method on the Sprite is not possible, at least not without rethinking the whole solution, so I would prefere to not add those. Let me know if you disagree

Reply:
> the static method on the Sprite is not possible

I don't see why that is the case. but it's definitely a separated method we could add later, so let's discuss that in a followup

> we have the variableSenquence and sequence, and that constructor was almost identical

yes and I hate that. I have some ideas for improving Animation constructors. I will put a followup after this with better constructors now that the image is always loaded.